scripts/olcf_summit_dev_cmake

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+EXTRA_ARGS=("$@")
+ARGS=(
+    -D CMAKE_BUILD_TYPE=Debug
+    -D BUILD_SHARED_LIBS=ON
+    ### COMPILERS AND FLAGS ###
+    -D CMAKE_CXX_FLAGS="-g -arch=sm_60 -lineinfo \
+        -std=c++11 --expt-extended-lambda \
+        -Xcudafe --diag_suppress=conversion_function_not_usable \
+        -Xcudafe --diag_suppress=cc_clobber_ignored \
+        -Xcudafe --diag_suppress=code_is_unreachable"
+    -D DataTransferKit_CXX_FLAGS="-Wall -Wno-shadow -Wpedantic"
+    ### TPLs ###
+    -D TPL_ENABLE_MPI=ON
+    -D TPL_ENABLE_BLAS=ON
+    -D TPL_ENABLE_LAPACK=ON
+    -D TPL_ENABLE_CUDA=ON
+    -D TPL_ENABLE_Boost=ON
+        -D Boost_INCLUDE_DIRS=$BOOST_DIR/include
+        -D Boost_LIBRARY_DIRS=$BOOST_DIR/lib
+    -D TPL_ENABLE_BoostLib=ON
+        -D BoostLib_INCLUDE_DIRS=$BOOST_DIR/include
+        -D BoostLib_LIBRARY_DIRS=$BOOST_DIR/lib
```

Comment:
I cannot remember what TriBITS does with `<TPL>_{INCLUDE,LIB}_DIRS`, if does prepend it to the search path or if it throw an error if the path provided does not contains whatever it is looking for.  (I suspect the former.)